vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/parser/json/JsonDefinitionParser.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2023 - 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.vssprocessor.parser.json
+
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+import org.eclipse.kuksa.vssprocessor.parser.VssDefinitionParser
+import org.eclipse.kuksa.vssprocessor.spec.VssSpecificationSpecModel
+import java.io.File
+import java.io.IOException
+
+private const val ROOT_KEY_VEHICLE = "Vehicle"
+
+private const val KEY_DATA_DESCRIPTION = "description"
+private const val KEY_DATA_TYPE = "type"
+private const val KEY_DATA_UUID = "uuid"
+private const val KEY_DATA_COMMENT = "comment"
+private const val KEY_DATA_DATATYPE = "datatype"
+private const val KEY_DATA_CHILDREN = "children"
+
+internal class JsonDefinitionParser : VssDefinitionParser {
+    private val dataKeys = listOf(
+        KEY_DATA_DESCRIPTION,
+        KEY_DATA_TYPE,
+        KEY_DATA_UUID,
+        KEY_DATA_COMMENT,
+        KEY_DATA_DATATYPE,
+        KEY_DATA_CHILDREN,
+    )
+
+    override fun parseSpecifications(definitionFile: File): List<VssSpecificationSpecModel> {
+        val vssSpecificationSpecModels = mutableListOf<VssSpecificationSpecModel>()
+
+        try {
+            val jsonStreamReader = definitionFile.reader()
+
+            val gson = Gson()
+            val rootJsonObject = gson.fromJson(jsonStreamReader, JsonObject::class.java)
+
+            if (rootJsonObject.has(ROOT_KEY_VEHICLE)) {
```

Comment:
Can we invert the if so the happy path is less nested?

Reply:
You kinda are like an if-else allergic. Please keep in mind that sometimes a clear "if-else" structure (if it's not too complex) is much nicer than using spartan programming with early exits. 

Personally I favor spartan programming if the blocks inside the if / else are many lines long and / or a really deep nesting can be avoided. In this case the branches are just a few lines long with a maximum method nesting of 2. 
We don't really save a nesting level because our max nesting level still stays 2 and is not reduced from 3 to 2 or something. 

So I prefer not to make this change in this case, because I think for the time being this is clearer than with an early exit

Reply:
OK for me after discussing it.